### PR TITLE
[main] allow empty string as first parameter to fetch

### PR DESCRIPTION
**Pull request by jorupp**
Allow calling `fetch` with an empty string as the first parameter.

Fixes #2419.

### CODE REVIEW

File: extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts [hunk 1]
```diff
@@ -1628,6 +1628,124 @@ export class AjaxTests extends AITestClass {
             }]
         });
 
+        this.testCaseAsync({
```
Comment (jorupp) [inline extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts:1631]: I cloned the two test cases above this that were testing for `null`/`undefined`, but I didn't completely understand what all of the setup/checks were that were necessary.

File: extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts [hunk 2]
```diff
@@ -1628,6 +1628,124 @@ export class AjaxTests extends AITestClass {
...
+                    // Assert
+                    Assert.ok(fetchSpy.calledOnce, "createFetchRecord called once after using fetch");
+                    let data = fetchSpy.args[0][0].baseData;
+                    Assert.equal("Fetch", data.type, "request is Fetch type");
+                    Assert.equal(false, throwSpy.called, "We should not have failed internally");
+                    Assert.equal(1, dependencyFields.length, "trackDependencyDataInternal was called");
+                    Assert.ok(dependencyFields[0].dependency.startTime, "startTime was specified before trackDependencyDataInternal was called");
+                    Assert.equal(undefined, dependencyFields[0].sysProperties, "no system properties");
+
+                    // Assert that the HTTP method was preserved
```
Comment (jorupp) [inline extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts:1674]: this block is new relative to the unit tests I copied this from.

**Comment by MSNev** [comment]
@jorupp can you please review and sign the CLA (if acceptable). I've already backported and committed your changes to `master`

**Comment by jorupp** [comment]
> @jorupp can you please review and sign the CLA (if acceptable). I've already backported and committed your changes to `master`

The guy from our legal team hasn't gotten back to me yet - I'm planning to bug him again in the morning (US CT).

**Comment by jorupp** [comment]
@microsoft-github-policy-service agree company="Rightpoint"

**Comment by jorupp** [comment]
> @jorupp can you please review and sign the CLA (if acceptable). I've already backported and committed your changes to `master`

@MSNev  the original comment from the CLA bot seems to have disappeared for me, but hopefully my previous comment will work for it.

**Review by MSNev — APPROVED** [review]

**Pull request merged**